molgenis-core-ui/src/main/java/org/molgenis/ui/admin/user/UserAccountController.java

```diff
@@ -37,26 +49,40 @@
 
 	private final UserAccountService userAccountService;
 	private final LanguageService languageService;
+	private final RecoveryService recoveryService;
+	private final TwoFactorAuthenticationService twoFactorAuthenticationService;
+	private final AppSettings appSettings;
 
 	@Autowired
-	public UserAccountController(UserAccountService userAccountService, LanguageService languageService)
+	public UserAccountController(UserAccountService userAccountService, LanguageService languageService,
+			RecoveryService recoveryService, TwoFactorAuthenticationService twoFactorAuthenticationService,
+			AppSettings appSettings)
 	{
 		super(URI);
 		this.userAccountService = requireNonNull(userAccountService);
 		this.languageService = requireNonNull(languageService);
+		this.recoveryService = requireNonNull(recoveryService);
+		this.twoFactorAuthenticationService = requireNonNull(twoFactorAuthenticationService);
+		this.appSettings = requireNonNull(appSettings);
 	}
 
-	@RequestMapping(method = RequestMethod.GET)
-	public String showAccount(Model model)
+	@RequestMapping(method = GET)
+	public String showAccount(Model model, @RequestParam(defaultValue = "false") boolean showCodes)
 	{
+		String twoFactorAuthenticationApp = appSettings.getTwoFactorAuthentication();
+		boolean twoFactorAuthenticationUser = userAccountService.getCurrentUser().getTwoFactorAuthentication();
+
 		model.addAttribute("user", userAccountService.getCurrentUser());
 		model.addAttribute("countries", CountryCodes.get());
 		model.addAttribute("groups", Lists.newArrayList(userAccountService.getCurrentUserGroups()));
 		model.addAttribute("min_password_length", MIN_PASSWORD_LENGTH);
+		model.addAttribute("two_factor_authentication_app", twoFactorAuthenticationApp);
```

Comment:
two_factor_authentication_app --> two_factor_authentication_app_option 
two_factor_authentication_app is an enum, that is why we changed the name from enabled to option